File: general/quicksort/quicksort.py
from abc import abstractproperty


def quicksort(arr):
  if len(arr) < 2:
    return arr
  else:
    pivot = arr[0]
    less = [i for i in arr[1:] if i <= pivot]
    greater = [i for i in arr[1:] if i > pivot]
    return quicksort(less) + [pivot] + quicksort(greater)

test1 = [10, 5, 2, 3]
test2 = [11, 2, 1002, 214, 4, 412]
print(quicksort(test1))
print(quicksort(test2))

File: general/quicksort/quicksort.js
const quicksort = (arr) => {
  if (arr.length < 2) {
    return arr;
  } else {
    let pivot = arr[0], less = [], greater = [];
    for (let i = 1; i < arr.length; i ++) {
      if (arr[i] <= pivot) less.push(arr[i]);
      else greater.push(arr[i]);
    }
    return quicksort(less).concat(pivot).concat(quicksort(greater));
  }
}

module.exports = quicksort;